lib/rules/no-empty.js

```diff
@@ -71,13 +74,42 @@ module.exports = {
                     return;
                 }
 
-                context.report({ node, messageId: "unexpected", data: { type: "block" } });
+                context.report({
+                    node,
+                    messageId: "unexpected",
+                    data: { type: "block" },
+                    suggest: [
+                        {
+                            messageId: "suggestComment",
+                            data: { type: "block" },
+                            fix(fixer) {
+                                return fixer.replaceTextRange(range, " /* empty */ ");
+                            }
+                        }
+                    ]
+                });
             },
 
             SwitchStatement(node) {
+                const closingCurly = sourceCode.getLastToken(node);
+                const openingCurly = sourceCode.getTokenBefore(node.cases.length ? node.cases[0] : closingCurly);
+                const range = [openingCurly.range[0] + 1, closingCurly.range[1] - 1];
 
                 if (typeof node.cases === "undefined" || node.cases.length === 0) {
-                    context.report({ node, messageId: "unexpected", data: { type: "switch" } });
+                    context.report({
+                        node,
+                        messageId: "unexpected",
+                        data: { type: "switch" },
+                        suggest: [
+                            {
+                                messageId: "suggestComment",
+                                data: { type: "switch" },
+                                fix(fixer) {
+                                    return fixer.replaceTextRange(range, " /* empty */ ");
+                                }
+                            }
+                        ]
```

Comment:
The implementation is correct, but adding a comment inside an empty SwitchStatement doesn't really help, because it's still considered an error by this rule ([playground link](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWVtcHR5OiBcImVycm9yXCIgKi9cblxuc3dpdGNoIChmb28pIHsgLyogZW1wdHkgKi8gfSAvLyBzdGlsbCBlcnJvciwgZW1wdHkgc3dpdGNoIGlzbid0IGFsbG93ZWQgcmVnYXJkbGVzcyBvZiBjb21tZW50cyBpbnNpZGVcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiZXM2Ijp0cnVlfX19)) so it seems we should provide these suggestions only for BlockStatement nodes.

Reply:
In other words, this rule shouldn't provide suggestion to add a comment inside an empty switch statement, because that won't fix the lint problem reported by this rule.

Reply:
I will update.